src/Dependencies/Collections/ImmutableSegmentedDictionary`2+Builder+PrivateMarshal.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Collections;
+
+internal readonly partial struct ImmutableSegmentedDictionary<TKey, TValue>
+{
+    public partial class Builder
+    {
+        /// <summary>
+        /// Private helper class for use only by <see cref="SegmentedCollectionsMarshal"/>.
+        /// </summary>
+        internal static class PrivateMarshal
+        {
+            /// <inheritdoc cref="SegmentedCollectionsMarshal.GetValueRefOrNullRef{TKey, TValue}(ImmutableSegmentedDictionary{TKey, TValue}.Builder, TKey)"/>
+            public static ref TValue FindValue(Builder dictionary, TKey key)
+                => ref SegmentedCollectionsMarshal.GetValueRefOrNullRef(dictionary.GetOrCreateMutableDictionary(), key);
+
+            /// <inheritdoc cref="SegmentedCollectionsMarshal.GetValueRefOrAddDefault{TKey, TValue}(ImmutableSegmentedDictionary{TKey, TValue}.Builder, TKey, out bool)"/>
+            public static ref TValue? GetValueRefOrAddDefault(Builder dictionary, TKey key, out bool exists)
```

Comment:
>GetValueRefOrAddDefault [](http://example.com/codeflow?start=38&length=23)

For the PR where I wanted this (72364), wouldn't GetValueRefOrNullRef have been sufficient as it just needs to be able to modify the colleciton while enumerating it?

Reply:
nm, I need to read a bit more before commenting as I now see the FindValue method above for that purpose